### PR TITLE
enable testing DDP examples

### LINKED ISSUE #4694: fix failing examples
## 🐛 Bug

We have uncommented PL examples in #4551 but two of them turned to be failing and shall be fixed:
```
FAILED pl_examples/test_examples.py::test_examples_dp_image_classifier[\n--max_epochs 1 --batch_size 32 --limit_train_batches 2 --limit_val_batches 2 --gpus 2 --distributed_backend dp --precision 16 ]
FAILED pl_examples/test_examples.py::test_examples_dp_autoencoder[\n--max_epochs 1 --batch_size 32 --limit_train_batches 2 --limit_val_batches 2 --gpus 2 --distributed_backend dp --precision 16 ]
```

### To Reproduce

uncomment two remaining examples with TODO comment

### Environment

Drone Multi-GPU testing

### Additional context

get back full example testing...

### PR DESCRIPTION
**Pull request by Borda**
## What does this PR do?

enable DDP test for example

Resolves #4694 

## Before submitting
- [x] Was this discussed/approved via a Github issue? (no need for typos and docs improvements)
- [x] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), Pull Request section?
- [x] Did you make sure your PR does only one thing, instead of bundling different changes together? Otherwise, we ask you to create a separate PR for every change.
- [ ] Did you make sure to update the documentation with your changes?
- [ ] Did you write any new necessary tests? 
- [ ] Did you verify new and existing tests pass locally with your changes?
- [ ] If you made a notable change (that affects users), did you update the [CHANGELOG](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/CHANGELOG.md)?

## PR review
Anyone in the community is free to review the PR once the tests have passed.
Before you start reviewing make sure you have read [Review guidelines](https://github.com/PyTorchLightning/pytorch-lightning/wiki/Review-guidelines). In short, see the following bullet-list:  

 - [x] Is this pull request ready for review? (if not, please submit in draft mode)
 - [x] Check that all items from **Before submitting** are resolved
 - [x] Make sure the title is self-explanatory and the description concisely explains the PR
 - [x] Add labels and milestones (and optionally projects) to the PR so it can be classified; _Bugfixes should be including in bug-fix release milestones (m.f.X) and features should be included in (m.X.b) releases._
 

## Did you have fun?
Make sure you had fun coding 🙃

### CODE REVIEW

**Comment by Borda** [comment]
seems that the DDP is hanging...

**Review by tchaton — APPROVED** [review]

**Review by tchaton — COMMENTED** [review]
Drone are breaking. When resolved, LGTM !

**Comment by awaelchli** [comment]
@Borda by how much will this increase drone test runtime?

**Comment by justusschock** [comment]
I think, we should only run this nightly with a cron job or so. Because when we run this more often, we will significantly slow down the development process, as we already experience some bottlenecks with iron and then it would pile up even more.

**Comment by Borda** [comment]
> I think, we should only run this nightly with a cron job or so. Because when we run this more often, we will significantly slow down the development process, as we already experience some bottlenecks with iron and then it would pile up even more.

well then you can say the same for each test we run, we can run a test just nightly :]
if you want some simplification we shall skip all code testing for pure doc changes...

**Review by tchaton — APPROVED** [review]
LGTM !

**Comment by justusschock** [comment]
> > I think, we should only run this nightly with a cron job or so. Because when we run this more often, we will significantly slow down the development process, as we already experience some bottlenecks with iron and then it would pile up even more.
> 
> well then you can say the same for each test we run, we can run a test just nightly :]
> if we want some simplification we shall skip all code testing for pure doc changes...

I don't think so. There's a difference in running unit tests and examples (which are basically some kind of integration/functionality testing).

**Comment by awaelchli** [comment]
I very much agree with Justus on this one

**Comment by Borda** [comment]
> I don't think so. There's a difference in running unit tests and examples (which are basically some kind of integration/functionality testing).

so you say we shall drop the test and simply call the script in a command-line as another entry after `python -m pytest`

**Comment by Borda** [comment]
@justusschock @awaelchli this is what you had in mind b26adda?

**Comment by awaelchli** [comment]
it looks like this would still run on every commit?

**Comment by Borda** [comment]
> it looks like this would still run on every commit/PR?

yes, or what is your suggestion, just at midnight, but then we do not know what PR breaks it... (well, stop it which would break it...)

**Comment by Borda** [comment]
@justusschock @awaelchli so whall we keep this DDP testing or just drop it to run it only on master?

**Comment by awaelchli** [comment]
running only on master would be wiser I think, given that some of these examples run for several seconds. Or you could also group them with the other "special tests" that we have.

**Review by tchaton — APPROVED** [review]

File: pl_examples/test_examples.py [hunk 1]
```diff
@@ -72,14 +67,13 @@ def test_examples_dp(tmpdir, import_cli, cli_args):
         module.cli_main()
 
 
-# ToDo: fix this failing example
 # @pytest.mark.parametrize('import_cli', [
```
Comment (tchaton) [inline pl_examples/test_examples.py:70]: Do we skip this file with `pl_examples/run_ddp-example.sh` ?

**Review by tchaton — APPROVED** [review]
LGTM !

**Review by carmocca — APPROVED** [review]

**Review by SeanNaren — APPROVED** [review]

**Pull request merged**